app/utils/service_client/middleware/jwt_authenticator.rb

```diff
@@ -1,21 +1,24 @@
 module ServiceClient
   module Middleware
 
+    IDENTITY = ->() {}
+
     AuthContext = EntityUtils.define_builder(
       [:marketplace_id, :uuid, :mandatory],
       [:actor_id, :uuid, :mandatory]
     )
 
     class JwtAuthenticator < MiddlewareBase
 
-      def initialize(disable_authentication, token_secret)
-        @_disabled = disable_authentication
-        @_secret = token_secret
+      def initialize(disable:, secret:, default_auth_context: IDENTITY)
+        @_disabled = disable
+        @_secret = secret
+        @_default_auth_context = default_auth_context
       end
 
       def enter(ctx)
         unless @_disabled
-          token = create_token(ctx[:opts][:auth_context])
+          token = create_token(ctx[:opts][:auth_context] || @_default_auth_context.call)
```

Comment:
I think we should have at least log a warning in case authentication is disabled but neither the parameters nor the auth context builder give us one. This can happen if session context is relied on but is not in place due to e.g. forgetting to include serialization mixin in a job.